### PR TITLE
feature: Support validation groups with `@Validated`

### PR DESCRIPTION
**Pull request by kevin-wise**
Enable specifying the validation groups to be used with `@Validated`. For example, `@Validated(groups = Group1.class)`. Should this be based on 3.5.x, since it's a new feature?

Also, this is my first time trying to update the documentation. Hopefully I got the syntax right.

### CODE REVIEW

**Comment by graemerocher** [comment]
Yes please target 3.5.x

**Comment by kevin-wise** [comment]
> Yes please target 3.5.x

ok. When will 3.4.x be merged into 3.5.x? This PR depends on the fix that was just merged to 3.4.x.  As soon as that merges I'll rebase.

**Comment by graemerocher** [comment]
I have merged up. Thanks.

**Comment by kevin-wise** [comment]
Rebased and base branch updated. Should be ready for review

**Comment by kevin-wise** [comment]
@graemerocher could I get a review? Would like to get this into 3.5.x if possible

File: validation/src/main/java/io/micronaut/validation/Validated.java
Comment (graemerocher) [inline validation/src/main/java/io/micronaut/validation/Validated.java:49]: Add `@since 3.5.0` to this method

File: validation/src/main/java/io/micronaut/validation/ValidatingInterceptor.java
Comment (graemerocher) [inline validation/src/main/java/io/micronaut/validation/ValidatingInterceptor.java]: Rewrite to `context.classeValues(Validated.class, "group")`

**Review by graemerocher — CHANGES_REQUESTED** [review]
Sorry for the delay, was away at a conference. Looks good but for a couple of minor things.

File: validation/src/main/java/io/micronaut/validation/ValidatingInterceptor.java
Comment (kevin-wise) [reply validation/src/main/java/io/micronaut/validation/ValidatingInterceptor.java]: done

File: validation/src/main/java/io/micronaut/validation/Validated.java
Comment (kevin-wise) [reply validation/src/main/java/io/micronaut/validation/Validated.java:49]: done

**Comment by kevin-wise** [comment]
updated as requested, and rebased. No problem on the delay, I understand. Just didn't know what to expect since I'm new to this project. I hope tagging you was ok.

**Comment by graemerocher** [comment]
tagging me is good, sometimes I need reminders :)

**Review by graemerocher — APPROVED** [review]

**Comment by kevin-wise** [comment]
sorry for the additional force pushes. Just fixing the format on the javadoc to match convention. All done now, promise :)

**Comment by kevin-wise** [comment]
@graemerocher does this need another review? Or can it get merged

**Review by sdelamo — APPROVED** [review]
Thanks for the contribution

**Pull request merged**
